lib/client.js

```diff
@@ -275,6 +275,71 @@ function Client(opts) {
       }, cb);
     }
   };
+  var deals = {
+    getRecentyModified: function(options, cb){
+      if (typeof options === 'function') {
+        cb = options;
+        options = {};
+      }
+      sendRequest({
+        method: 'GET',
+        url: self.baseUrl + '/deals/v1/deal/recent/modified',
+        qs: options
+      }, cb);
+    },
+    getRecentlyCreated: function(options, cb){
+      if(typeof options === 'function'){
+        cb = options;
+        options = {};
+      }
+      sendRequest({
+        method: 'GET',
+        url: self.baseUrl + '/deals/v1/deal/recent/created',
+        qs: options
+      }, cb);
+    },
+    getById: function(id, cb){
+      sendRequest({
+        method: 'GET',
+        url: self.baseUrl + '/deals/v1/deal/' + id
+      }, cb)
+    },
+    deleteById: function(id, cb){
+      sendRequest({
+        method: 'DELETE',
+        url: self.baseUrl + '/deals/v1/deal/' + id
+      }, cb);
+    },
+    updateById: function(id, data, cb){
+      sendRequest({
+        method: 'PUT',
+        url: self.baseUrl + '/deals/v1/deal/' + id,
+        body: data,
+      }, cb)
+    },
+    create: function(id, data, cb){
+      sendRequest({
+        method: 'POST',
+        url: self.baseUrl + ' /deals/v1/deal/',
+        body: data
+      }, cb)
+    },
+    associate: function(id, data, cb){
+      this.options = data;
+      sendRequest({
+        method: 'PUT',
+        url: self.baseUrl + 'deals/v1/deal/' + id + '/associations/'+ this.options.objectiveType +'?id=' + this.options.id,
```

Comment:
A couple of things
- I think we'll need a leading slash before `deals/v1/deal`
- I would change the function signature to something like `function(id, objectType, associatedObjectId, cb)` and remove `this.options` since we require these parameters and know what they are